Fix misc typos

pieterocp opened the pull request:
Ended up reading through the repo since I was having issues with `change_in`, it was an issue on my end since I didn't realise that all jobs will run if the files change, so read through [`changein/function.go`](https://github.com/semaphoreci/spc/blob/master/pkg/when/changein/function.go) and nosey me sees a typo, so trawled through and addressed them.

Nothing major, just internal changes, docs improvements, error messages made clearer.

pieterocp committed:
Fix misc typos

pieterocp committed:
Fix typo Requirments -> Requirements

pieterocp committed:
Fix typo Macthed -> Matched

pieterocp commented:
Any semaphore Devs able to give this a review?

hamir-suspect commented:
@pieterocp would you be willing to update the Makefile as well to use the publicly available security-toolbox like here: https://github.com/semaphoreci/cli/blob/master/Makefile#L9

pieterocp committed:
chore: use the public security toolbox over private repo

since this has been migrated, and now the private project doesn't exist

(based on PR feedback + https://github.com/semaphoreci/cli/pull/239/files)

lucianoliberti commented:
@pieterocp, you need to rebase.

pieterocp commented:
> @pieterocp, you need to rebase.


Done. @lucianoliberti It would appreciated if this is merged in soon before it requires yet another master merge / rebase.

loadez commented:
/sem-approve

skipi commented:
/sem-approve

loadez commented:
Hey @pieterocp, sorry that this is taking more time than it's needed. As soon as we solve the stuck check I will notify you so we can merge this pull request.

And thank you for submitting it.

pieterocp commented:
> Hey @pieterocp, sorry that this is taking more time than it's needed. As soon as we solve the stuck check I will notify you so we can merge this pull request.
> 
> And thank you for submitting it.

Maybe you should try GitHub actions or CircleCI 😋😂🤣

No worries at all, it's all good 👍
